src/generators/java/JavaConstrainer.ts

```diff
@@ -74,11 +74,48 @@ const interpretUnionValueType = (types: string[]): string => {
   return 'Object';
 };
 
+function unionIncludesBuiltInTypes(model: ConstrainedUnionModel): boolean {
+  if (model.union instanceof Boolean) {
+    return true;
+  }
+  const builtInType = model.union.find(
+    (u) =>
+      //Primitive types and boxed versions
+      u.type.toLowerCase() === 'byte' ||
+      u.type.toLowerCase() === 'short' ||
+      u.type === 'int' ||
+      u.type === 'Integer' ||
+      u.type.toLowerCase() === 'long' ||
+      u.type.toLowerCase() === 'float' ||
+      u.type.toLowerCase() === 'double' ||
+      u.type === 'char' ||
+      u.type === 'Character' ||
+      u.type.toLowerCase() === 'boolean' ||
+      u.type.toLowerCase() === 'int' ||
+      u.type === 'Object' ||
+      u.type === 'String' ||
+      //Collections
+      u.type.startsWith('Map<') ||
+      u.type.startsWith('List<') ||
+      u.type.startsWith('Set<') ||
+      u.type.endsWith('[]')
+  );
+  return builtInType !== undefined;
+}
```

Comment:
what if you change this to check for only object models instead? You can use this function in `JavaGenerator.renderUnion` to not render the interface to avoid rendering useless interfaces. 

```ts
export function unionIncludesBuiltInTypes(
  model: ConstrainedUnionModel
): boolean {
  return !model.union.every(
    (union) =>
      union instanceof ConstrainedObjectModel ||
      (union instanceof ConstrainedReferenceModel &&
        union.ref instanceof ConstrainedObjectModel)
  );
}
```

Reply:
Thanks. Hopefully will find the time to test out these changes sometime today/this week.

Reply:
Swapped out `unionIncludesBuiltInTypes` implementation to use modelina types. Will look at supressing the interface generation next.

Reply:
Interfaces for non `ConstrainedObjectModel` unions are now supressed.